addons/sale/models/payment.py

```diff
@@ -61,33 +61,40 @@ def _set_transaction_pending(self):
         super(PaymentTransaction, self)._set_transaction_pending()
 
         for record in self:
-            sales_orders = record.sale_order_ids.filtered(lambda so: so.state == 'draft')
-            sales_orders.force_quotation_send()
+            sales_orders = record.sale_order_ids.filtered(lambda so: so.state in ['draft', 'sent'])
+            # set quote as sent when transaction is pending
+            # but do not send mail because mail will be send based on configuration in setting.
+            sales_orders.with_context(tracking_disable=True).write({'state': 'sent'})
 
             if record.acquirer_id.provider == 'transfer':
                 for so in record.sale_order_ids:
                     so.reference = record._compute_sale_order_reference(so)
+            # send order confirmation mail
+            sales_orders.send_order_confirmation_mail()
 
     @api.multi
     def _set_transaction_authorized(self):
         # Override of '_set_transaction_authorized' in the 'payment' module
         # to confirm the quotations automatically.
         super(PaymentTransaction, self)._set_transaction_authorized()
-        sales_orders = self.mapped('sale_order_ids').filtered(lambda so: so.state == 'draft')
-        sales_orders.force_quotation_send()
-        sales_orders = self.mapped('sale_order_ids').filtered(lambda so: so.state == 'sent')
+        sales_orders = self.mapped('sale_order_ids').filtered(lambda so: so.state in ['draft', 'sent'])
```

Comment:
Same here, the previous code only sent emails for sale_orders that are in "draft".
Here you will send emails for both "draft" and "sent" sale_orders, why ?

Reply:
@awa-odoo , same as above. Thanks